examples/catalog/lib/custom_a11y_traversal.dart

```diff
@@ -0,0 +1,342 @@
+    case Field.DOGS:
+      return 'Dogs';
+    case Field.CATS:
+      return 'Cats';
+    case Field.FISH:
+    default:
+      return 'Fish';
+  }
+}
+
+// --------------- Main app ---------------------
+
+/// The top-level example widget that serves as the body of the app.
+class CustomTraversalExample extends StatefulWidget {
+  @override
+  CustomTraversalExampleState createState() => new CustomTraversalExampleState();
+}
+
+/// The state object for the top level example widget.
+class CustomTraversalExampleState extends State<CustomTraversalExample> {
+  /// The fields that we are manipulating. List indices correspond to
+  /// the entries in the [Field] enum.
+  List<int> fields = <int>[0, 0, 0];
+
+  void _addToField(Field field, int delta) {
+    setState(() {
+      fields[field.index] += delta;
+    });
+  }
+
+  Widget _makeFieldHeader(int rowOrder, int columnOrder, Field field) {
+    return new RowColTraversal(
+      rowOrder: rowOrder,
+      columnOrder: columnOrder,
+      child: new Text(_fieldToName(field), textScaleFactor: 2.0),
+    );
+  }
+
+  Widget _makeSpinnerButton(int rowOrder, int columnOrder, Field field, {bool increment: true}) {
+    return new SpinnerButton(
+      rowOrder: rowOrder,
+      columnOrder: columnOrder,
+      icon: increment ? Icons.arrow_upward : Icons.arrow_downward,
+      onPressed: () => _addToField(field, increment ? 1 : -1),
+      field: field,
+      increment: increment,
+    );
+  }
+
+  Widget _makeEntryField(int rowOrder, int columnOrder, Field field) {
+    return new FieldWidget(
+      rowOrder: rowOrder,
+      columnOrder: columnOrder,
+      onIncrease: () => _addToField(field, 1),
+      onDecrease: () => _addToField(field, -1),
+      value: fields[field.index],
+      field: field,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      home: new Scaffold(
+        appBar: new AppBar(
+          title: const Text('Pet Inventory'),
+        ),
+        body: new Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: <Widget>[
+            new Semantics(
+              // Since this is the only sort key that the text has, it
+              // will be compared with the 'column' OrdinalSortKeys of all the
+              // fields, because the column sort keys are first in the other fields.
+              //
+              // an ordinal of "0.0" means that it will be traversed before column 1.
+              sortKey: const OrdinalSortKey(0.0),
+              child: const Text(
+                'How many pets do you own?',
+                textScaleFactor: 2.0,
+              ),
+            ),
+            const Padding(padding: const EdgeInsets.symmetric(vertical: 10.0)),
+            new Row(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              children: <Widget>[
+                _makeFieldHeader(1, 0, Field.DOGS),
```

Comment:
nit: I believe this would be easier to read if the int params are (required) named arguments. That way, I don't have to remember the order column and row params are specified in.

(here and below)